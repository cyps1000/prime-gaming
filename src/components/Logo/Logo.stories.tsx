/**
 * Storybook
 * @see https://storybook.js.org/docs/react/get-started/introduction
 */
import { Story, Meta } from "@storybook/react";

/**
 * Component Imports
 */
import Logo, { LogoProps } from "./Logo";

export default {
  title: "Components/Logo",
  component: Logo,
} as Meta;

/**
 * Defines the Template
 * @param args LogoProps
 * @returns
 */
const Template: Story<LogoProps> = (args) => (
  <Logo {...args} />
);

/**
 * Default case
 */
export const Default = Template.bind({});

/**
 * Put your component props in here
 *
 * Example:
 * Default.args = {
 *  text: "Hello World"
 * }
 *
 * Assuming that the component expects a prop text that is a string.
 */
Default.args = {};
