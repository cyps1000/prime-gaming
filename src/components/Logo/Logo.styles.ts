/**
 *  Material UI Imports
 * @see https://material-ui.com/styles/basics/
 */
import { makeStyles } from "@material-ui/core/styles";

/**
 * Styles the component
 */
const useStyles = makeStyles((theme) => ({
  logo: {
    maxWidth: 150,
    marginTop: -5,
  },
}));

export { useStyles };
