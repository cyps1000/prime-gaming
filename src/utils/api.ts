import axios from "axios";

/**
 * Defines the hook props interface
 */
interface Config {
  mock: boolean;
}

export const getApiClient = (config: Config) => {
  const { mock } = config;

  /**
   * Handles getting the base api url
   */
  const getApiUrl = () => {
    if (process.env.NODE_ENV === "development") {
      if (mock) return "/";
      return process.env.REACT_APP_LOCAL_API;
    }
    return process.env.REACT_APP_STAGING_API;
  };

  const apiClient = axios.create({
    baseURL: getApiUrl(),
  });

  return { apiClient };
};
