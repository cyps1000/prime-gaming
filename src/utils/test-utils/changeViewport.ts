/**
 * External imports
 */
import mediaQuery from "css-mediaquery";

/**
 * Defines the BreakpointKey type
 */
type BreakpointKey = "xl" | "lg" | "md" | "sm" | "xs";

/**
 * Handles getting the width based on the breakpoint key provided
 */
const getWidthByThemeBreakpoints = (breakpointKey: BreakpointKey) => {
  switch (breakpointKey) {
    case "xl":
      return 1920;
    case "lg":
      return 1280;
    case "md":
      return 960;
    case "sm":
      return 600;
    case "xs":
      return 414;
    default:
      return 1920;
  }
};

/**
 * Handles changing the viewport
 */
export const changeViewport = (breakpointKey: BreakpointKey): any => {
  let width: number = getWidthByThemeBreakpoints(breakpointKey);

  return (query: string) => {
    return {
      matches: mediaQuery.match(query, { width: width }),
      addListener: () => {},
      removeListener: () => {},
    };
  };
};
