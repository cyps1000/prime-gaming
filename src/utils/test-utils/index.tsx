import React, { ReactElement } from "react";
/**
 * External Imports
 */
import { MemoryRouter } from "react-router-dom";

/**
 * Imports testing utils
 */
import { render, RenderOptions } from "@testing-library/react";

/**
 * Imports providers
 */

import { ThemeProvider, LanguageProvider } from "../../hooks";
import Providers from "../../components/Providers";

/**
 * Defines the Wrapper component
 */
const Wrapper: React.FC = ({ children }) => {
  return (
    <ThemeProvider>
      <LanguageProvider>
        <Providers>
          <MemoryRouter>{children}</MemoryRouter>
        </Providers>
      </LanguageProvider>
    </ThemeProvider>
  );
};

/**
 * Defines the custom render
 */
const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: Wrapper, ...options });

/**
 * Re-exports everything from the testing library
 */
export * from "@testing-library/react";

export * from "./changeViewport";
export * from "./mocks";
export { default as userEvent } from "@testing-library/user-event";
export { customRender as render };
