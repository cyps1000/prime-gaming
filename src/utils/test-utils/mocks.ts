export const mockHistoryPush = jest.fn();
