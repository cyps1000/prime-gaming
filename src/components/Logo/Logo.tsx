/**
 * External Imports
 */
import clsx from "clsx";

/**
 * Imports assets
 */
import logo from "./logo.png";

/**
 * Imports the component styles
 */
import { useStyles } from "./Logo.styles";

/**
 * Defines the props interface
 */
export interface LogoProps {
  className?: string | any;
}

/**
 * Displays the component
 */
const Logo: React.FC<LogoProps> = (props) => {
  const { className } = props;

  /**
   * Gets the component styles
   */
  const classes = useStyles();

  /**
   * Defines the component classes
   */
  const componentClasses = clsx(classes.logo, {
    [className]: !!className,
  });

  return <img className={componentClasses} src={logo} alt="logo" />;
};

export default Logo;
