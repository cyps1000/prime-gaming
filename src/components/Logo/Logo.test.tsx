/**
 * @see https://testing-library.com/docs/react-testing-library/intro
 * @see https://www.robinwieruch.de/react-testing-library
 * @see https://www.smashingmagazine.com/2020/07/react-apps-testing-library/
 */

/**
 * Imports test utils
 */
import { render, fireEvent } from "../../utils/test-utils";
import { mockHistoryPush } from "../../utils/test-utils/mocks";

/**
 * External Imports
 */
import pretty from "pretty";

/**
 * Imports component
 */
import Logo from "./Logo";

/**
 * Logo rendering tests
 */
describe("Logo Rendering Tests", () => {
  it("renders the component without errors", () => {
    render(<Logo />);
  });

  it("renders an image with an alt of Logo", () => {
    const { getByAltText } = render(<Logo />);
    expect(getByAltText(/logo/i)).toBeInTheDocument();
  });

  it("has a default className", () => {
    const { container } = render(<Logo />);
    expect(container.firstChild).toHaveAttribute("class");
  });

  it("accepts a className prop and it adds it to the img tag", () => {
    const { container } = render(<Logo className="custom-logo" />);
    expect(container.firstChild).toHaveClass("custom-logo");
  });
});

/**
 * Logo logic tests
 */
describe("Logo Logic Tests", () => {
  it("takes the user to the home page when clicked", () => {
    const { getByAltText } = render(<Logo />);

    /**
     * Clicks on the logo
     */
    fireEvent.click(getByAltText("logo"));

    expect(mockHistoryPush).toHaveBeenCalledWith("/");
  });
});

/**
 * Logo snapshot test
 */
describe("Logo Snapshot Test", () => {
  it("passes the snapshot test", () => {
    const { container } = render(<Logo />);

    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });
});
