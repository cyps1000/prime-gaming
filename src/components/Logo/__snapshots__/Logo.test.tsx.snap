// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Snapshot Test passes the snapshot test 1`] = `"<img class=\\"makeStyles-logo-6\\" src=\\"logo.png\\" alt=\\"logo\\">"`;
